modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/input/stream/join/JoinProcessor.java

```diff
@@ -188,4 +186,25 @@ public void setStateEventPool(StateEventPool stateEventPool) {
         this.stateEventPool = stateEventPool;
     }
 
+    /**
+     * Join the given two event streams
+     * @param stream1 event stream 1
+     * @param stream2 event stream 2
+     */
+    public void joinBuilder(StreamEvent stream1,StreamEvent stream2){
+        StateEvent returnEvent = stateEventPool.borrowEvent();
+        returnEvent.setEvent(0, stream1);
+        returnEvent.setEvent(1, stream2);
+        if (preJoinProcessor) {
+            returnEvent.setType(ComplexEvent.Type.CURRENT);
+        } else {
+            returnEvent.setType(ComplexEvent.Type.EXPIRED);
+        }
+        if (!leftJoinProcessor) {
+            returnEvent.setTimestamp(stream2.getTimestamp());
+        }else{
+            returnEvent.setTimestamp(stream1.getTimestamp());
+        }
+        returnEventChunk.add(returnEvent);
```

Comment:
adding to returnEventChunk should be out of the function joinBuilder, and join builder should return the built event. Because the function name does give a hint that it will be added to the return event chunk.